mod_reforged/hooks/skills/actives/hand_to_hand.nut

```diff
@@ -3,11 +3,11 @@
 	{
 		local ret = __original();
 
-		for (local index = (ret.len() - 1); index >= 0; index--)
+		foreach (i, entry in ret)
```

Comment:
I chose the decrementing for loop just to be future proof in case we want to change/remove another entry line.

Reply:
Alright I switched it to a decrementing for loop.

Reply:
The decrementing one has a catch though and that is if a mod/someone else adds a tooltip entry with id 6 then it will remove that one instead. So we also need to check for the icon as you had done earlier - but that isn't safe either as someone could add an entry with id 6 and hitchance icon different from the one we are intending to remove. So, maybe the incrementing loop is better?

Reply:
We can't put any other operations in that for loop like additional removal or changes of other tooltips. So that whole for loop at that point might aswell be a global utility `removeTooltipLine` function which takes a tooltip and then up to three additional arguments for id, icon and text.

Even still, an incrementing for loop will also fail when a mod adds new global tooltip lines to the `getDefaultTooltip` function in the skill.nut`, with the id 6.

Reply:
> We can't put any other operations in that for loop like additional removal or changes of other tooltips. So that whole for loop at that point might aswell be a global utility removeTooltipLine function which takes a tooltip and then up to three additional arguments for id, icon and text.

I don't know if we need a helper function for this but maybe it'd be useful. Seems like it is a rather niche use case and if you want to replace multiple entries then it is much more efficient to use a single loop rather than multiple calls to a function like that.

> Even still, an incrementing for loop will also fail when a mod adds new global tooltip lines to the getDefaultTooltip function in the skill.nut, with the id 6.

I think it is reasonable to say that this is a much less likely scenario than someone adding additional entries to the tooltip of individual skills at the bottom with the id 6.

In any case, it doesn't really matter to "pre-emptively optimize" for future potential scenarios. The most stable solution at this point is going to be one with an incrementing loop and if/when we want to remove more entries in the future we can switch to a decrementing one.